source/ada/lsp-ada_documents.adb

```diff
@@ -677,15 +708,141 @@ package body LSP.Ada_Documents is
          when Ada_For_Loop_Var_Decl |
               Ada_Label_Decl |
               Ada_Named_Stmt_Decl =>
-            return LSP.Messages.A_Constant;
+            return (if Ignore_Local
+                    then LSP.Messages.A_Null
+                    else LSP.Messages.A_Constant);
 
          when others
             => null;
       end case;
 
-      return LSP.Messages.A_Function;
+      return LSP.Messages.A_Null;
    end Get_Decl_Kind;
 
+   -----------------
+   -- Get_Profile --
+   -----------------
+
+   function Get_Profile
+     (Node : Libadalang.Analysis.Basic_Decl) return LSP.Types.LSP_String
+   is
+      use Ada.Strings.Wide_Wide_Unbounded;
+      use Libadalang.Analysis;
+      use Libadalang.Common;
+
+      function To_Text (Node : Ada_Node'Class) return Wide_Wide_String;
+      --  Retrieve the node text and format it
+
+      function To_Profile
+        (Node : Libadalang.Analysis.Subp_Spec'Class)
+         return LSP.Types.LSP_String;
+
+      -------------
+      -- To_Text --
+      -------------
+
+      function To_Text (Node : Ada_Node'Class) return Wide_Wide_String
```

Comment:
Why not return LSP_String directly?

Reply:
It's easier to concatenate Wide_Wide_String than LSP_String